packages/ember-htmlbars/tests/helpers/if_unless_test.js

```diff
@@ -51,248 +47,6 @@ QUnit.module('ember-htmlbars: {{#if}} and {{#unless}} helpers', {
-    array: array,
-
-    template: compile('{{#if view.array}}Yep{{/if}}')
-  });
-
-  runAppend(view);
-
-  equal(view.$().text(), '');
-
-  run(function() {
-    view.get('array').pushObject(1);
-  });
-
-  equal(view.$().text(), 'Yep');
-
-  run(function() {
-    view.get('array').removeObject(1);
-  });
-
-  equal(view.$().text(), '');
-}
-
-QUnit.test('The `if` helper updates if an array is empty or not', function() {
-  testIfArray(emberA());
-});
-
-QUnit.test('The `if` helper updates if an array-like object is empty or not', function() {
-  testIfArray(ArrayProxy.create({ content: emberA() }));
-});
-
-QUnit.test('The `unless` helper updates if an array-like object is empty or not', function() {
-  view = EmberView.create({
-    array: ArrayProxy.create({ content: emberA() }),
-
-    template: compile('{{#unless view.array}}Yep{{/unless}}')
-  });
-
-  runAppend(view);
-
-  equal(view.$().text(), 'Yep');
-
-  run(function() {
-    view.get('array').pushObject(1);
-  });
-
-  equal(view.$().text(), '');
-
-  run(function() {
-    view.get('array').removeObject(1);
-  });
-
-  equal(view.$().text(), 'Yep');
-});
-
-QUnit.test('The `if` helper updates when the value changes', function() {
-  view = EmberView.create({
-    conditional: true,
-    template: compile('{{#if view.conditional}}Yep{{/if}}')
-  });
-  runAppend(view);
-  equal(view.$().text(), 'Yep');
-  run(function() {
-    view.set('conditional', false);
-  });
-  equal(view.$().text(), '');
-});
-
-QUnit.test('The `if (unbound` helper does not update when the value changes', function() {
-  view = EmberView.create({
-    conditional: true,
-    template: compile('{{#if (unbound view.conditional)}}Yep{{/if}}')
-  });
-  runAppend(view);
-  equal(view.$().text(), 'Yep');
-  run(function() {
-    view.set('conditional', false);
-  });
-  equal(view.$().text(), 'Yep');
-});
-
-QUnit.test('The `unless` helper updates when the value changes', function() {
-  view = EmberView.create({
-    conditional: false,
-    template: compile('{{#unless view.conditional}}Nope{{/unless}}')
-  });
-  runAppend(view);
-  equal(view.$().text(), 'Nope');
-  run(function() {
-    view.set('conditional', true);
-  });
-  equal(view.$().text(), '');
-});
-
-QUnit.test('The `if (unbound` helper does not update when the value changes', function() {
-  view = EmberView.create({
-    conditional: false,
-    template: compile('{{#unless (unbound view.conditional)}}Nope{{/unless}}')
-  });
-  runAppend(view);
-  equal(view.$().text(), 'Nope');
-  run(function() {
-    view.set('conditional', true);
-  });
-  equal(view.$().text(), 'Nope');
-});
-
-QUnit.test('The `if (unbound` helper should work when its inverse is not present', function() {
-  view = EmberView.create({
-    conditional: false,
-    template: compile('{{#if (unbound view.conditional)}}Yep{{/if}}')
-  });
-  runAppend(view);
-  equal(view.$().text(), '');
-});
-
-QUnit.test('should not rerender if truthiness does not change', function() {
```

Comment:
Discussed with @rwjblue, removed without replacement. The test in its current form does not test anything interesting – the original test looks something like [this](https://github.com/emberjs/ember.js/commit/1cd1ae025387c3bb5a77d4ec27f4fb2b7ffe6349#diff-1642c5e5aaa2e35212a7580fdb4327deR226), which is actually testing something useful. However, we thought about the equivalent semantics in current Ember and not sure if there anything particularly important that we should test.